packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -308,16 +309,18 @@ class AppBar extends StatelessWidget {
         child: leading
       ));
     }
+
     toolBarRow.add(new Flexible(
-      child: new Align(
-        // TODO(abarth): In RTL this should be aligned to the right.
-        alignment: FractionalOffset.centerLeft,
-        child: new Padding(
-          padding: new EdgeInsets.only(left: 8.0),
-          child: effectiveCenterTitle ? null : centerWidget
+      child: new Padding(
+        padding: new EdgeInsets.symmetric(horizontal: 8.0),
+        child: new Align(
```

Comment:
I think we need a custom layout to get the title centering correct.

I think the correct algorithm is to place the left and right widgets, see how much space is left, and ask the title to layout with that much space.  Once you know the size of the title widget, we need to position it as close to center as possible without overlapping the left and right widgets.